client/storage/index-new.js

```diff
@@ -0,0 +1,436 @@
+const FormData = require('form-data');
+const axios = require("axios");
+
+// TODO: for some reason docker fails to resolve module if I move it to another file
+const getDownloadQuery = (chunkId, majorVersion, minorVersion) =>
+    gql`{
+      transactions(
+        tags: [
+          {
+            name: "__pn_chunk_${majorVersion}.${minorVersion}_id",
+            values: ["${chunkId}"]
+          }
+        ]
+      ) {
+        edges {
+          node {
+            id
+          }
+        }
+      }
+    }`;
+
+const DOWNLOAD_STATUS = {
+    NOT_STARTED: "NOT_STARTED",
+    IN_PROGRESS: "IN_PROGRESS",
+    COMPLETED: "COMPLETED",
+    FAILED: "FAILED"
+};
+
+const CHUNKINFO_PROLOGUE = "PN^CHUNK\x05$\x06z\xf5*INFO";
+const CONCURRENT_DOWNLOAD_DELAY = 100;
+
+// TODO: get rid of this and read config directly.
+// TODO: possibly split this file into several ones after refactoring config
+let config;
+let logger;
+let cacheDir;
+let filesDir;
+const init = async ctx => {
+    // TODO: ambiguous stuff, config is spread between storage and client.storage
+    config = {
+        ...ctx.config.storage,
+        ...ctx.config.client.storage
+    };
+    if (!config.arweave_experiment_version_major) {
+        throw new Error('arweave_experiment_version_major not set or is 0');
+    }
+    if (!config.arweave_experiment_version_minor) {
+        throw new Error('arweave_experiment_version_minor not set or is 0');
+    }
+    logger = ctx.log;
+    cacheDir = path.join(ctx.datadir, ctx.config.client.storage.cache_path);
+    // TODO: this should also come from config, but we don't have such a property
+    filesDir = path.join(ctx.datadir, "files");
+    await Promise.all([
+        makeSurePathExistsAsync(cacheDir),
+        makeSurePathExistsAsync(filesDir)
+    ]);
+};
+
+const arweave = Arweave.init({
+    port: 443,
+    protocol: 'https',
+    host: 'arweave.net'
+});
+
+const getChunk = async (chunkId, encoding = "utf8", useCache = true) => {
+    const chunk = await Chunk.findByIdOrCreate(chunkId);
+    const chunkPath = path.join(cacheDir, `chunk_${chunkId}`);
+
+    if (useCache && chunk.dl_status === DOWNLOAD_STATUS.COMPLETED) {
+        logger.debug(`Returning chunk ${chunkId} from cache`);
+        return fs.readFile(chunkPath, {encoding});
+    }
+    if (chunk.dl_status === DOWNLOAD_STATUS.IN_PROGRESS) {
+        logger.debug(`Chunk ${chunkId} download already in progress, waiting`);
+        await delay(CONCURRENT_DOWNLOAD_DELAY);
+        return getChunk(chunkId, encoding); // use cache should be true in this case
+    }
+
+    logger.debug(`Downloading chunk ${chunk.id}`);
+    chunk.dl_status = DOWNLOAD_STATUS.IN_PROGRESS;
+    await chunk.save();
+
+    try {
+        const query = getDownloadQuery(
+            chunkId,
+            config.arweave_experiment_version_major,
+            config.arweave_experiment_version_minor
+        );
+
+        const queryResult = await request('https://arweave.net/graphql', query);
```

Comment:
There is no such URL anywhere in the config. I've put TODO, my suggestion is not to touch this unless we get to the config refactor.

Reply:
> There is no such URL anywhere in the config. I've put TODO, my suggestion is not to touch this unless we get to the config refactor.

I mean, please add this url to the `defaultConfg.json` file next to `arweave_airdrop_endpoint` and get it from the config here instead of using the hardcoded string. Otherwise it cannot be changed when running the app in another environment (which I am already working on).